Add UCSCXenaShiny

ShixiangWang opened the pull request:
<!--
Thank you very much for putting in this recipe PR!

This repository is very active, so if you need help with
a PR or once it's ready for review, please let the right people know.
There are language-specific teams for reviewing recipes.

Currently available teams are:
- python `@conda-forge/help-python`
- python/c hybrid `@conda-forge/help-python-c`
- r `@conda-forge/help-r`
- java `@conda-forge/help-java`
- nodejs `@conda-forge/help-nodejs`
- c/c++ `@conda-forge/help-c-cpp`
- perl `@conda-forge/help-perl`
- Julia `@conda-forge/help-julia`
- ruby `@conda-forge/help-ruby`

If your PR doesn't fall into those categories please contact
the full review team `@conda-forge/staged-recipes`.

Due to GitHub limitations first time contributors to conda-forge are unable
to ping these teams. You can [ping the team][1] using a special command in
a comment on the PR to get the attention of the `staged-recipes` team. You can
also consider asking on our [Gitter channel][2] if your recipe isn't reviewed promptly.

[1]: https://conda-forge.org/docs/maintainer/infrastructure.html#conda-forge-admin-please-ping-team
[2]: https://gitter.im/conda-forge/conda-forge.github.io
-->

Checklist
- [x] Title of this PR is meaningful: e.g. "Adding my_nifty_package", not "updated meta.yaml".
- [x] License file is packaged (see [here](https://github.com/conda-forge/staged-recipes/blob/master/recipes/example/meta.yaml#L57-L66) for an example).
- [x] Source is from official source.
- [x] Package does not vendor other packages. (If a package uses the source of another package, they should be separate packages or the licenses of all packages need to be packaged).
- [x] If static libraries are linked in, the license of the static library is packaged.
- [x] Build number is 0.
- [x] A tarball (`url`) rather than a repo (e.g. `git_url`) is used in your recipe (see [here](https://conda-forge.org/docs/maintainer/adding_pkgs.html#build-from-tarballs-not-repos) for more details).
- [x] GitHub users listed in the maintainer section have posted a comment confirming they are willing to be listed there.
- [x] When in trouble, please check our [knowledge base documentation](https://conda-forge.org/docs/maintainer/knowledge_base.html) before pinging a team.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/ucscxenashiny```) and found some lint.

Here's what I've got...


For **recipes/ucscxenashiny**:

 * There are too few lines.  There should be one empty line at the end of the file.

For **recipes/ucscxenashiny**:

 * License is not an SPDX identifier (or a custom LicenseRef) nor an SPDX license expression.

Documentation on acceptable licenses can be found [here]( https://conda-forge.org/docs/maintainer/adding_pkgs.html#spdx-identifiers-and-expressions ).

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/ucscxenashiny```) and found some lint.

Here's what I've got...


For **recipes/ucscxenashiny**:

 * There are 1 too many lines.  There should be one empty line at the end of the file.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/ucscxenashiny```) and found it was in an excellent condition.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/ezcox```, ```recipes/ucscxenashiny```, ```recipes/ucscxenatools```) and found some lint.

Here's what I've got...


For **recipes/ezcox**:

 * The recipe could do with some maintainers listed in the `extra/recipe-maintainers` section.

For **recipes/ucscxenatools**:

 * The recipe could do with some maintainers listed in the `extra/recipe-maintainers` section.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/ezcox```, ```recipes/ucscxenashiny```, ```recipes/ucscxenatools```) and found it was in an excellent condition.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/ezcox```, ```recipes/forestmodel```, ```recipes/ucscxenashiny```, ```recipes/ucscxenatools```) and found it was in an excellent condition.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/ezcox```, ```recipes/forestmodel```, ```recipes/ucscxenashiny```, ```recipes/ucscxenatools```) and found some lint.

Here's what I've got...


For **recipes/ucscxenashiny**:

 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [34, 35, 53, 54]
 * `noarch` packages can't have selectors. If the selectors are necessary, please remove `noarch: generic`.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I wanted to let you know that I linted all conda-recipes in your PR (```recipes/ezcox```, ```recipes/forestmodel```, ```recipes/ucscxenashiny```, ```recipes/ucscxenatools```) and found some lint.

Here's what I've got...


For **recipes/ucscxenashiny**:

 * Selectors are suggested to take a ``<two spaces>#<one space>[<expression>]`` form. See lines [33, 34]

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/ezcox```, ```recipes/forestmodel```, ```recipes/ucscxenashiny```, ```recipes/ucscxenatools```) and found it was in an excellent condition.

ShixiangWang commented:
@conda-forge-admin, please ping team

The build failed due to colordiff，i dont know why？any suggestion？

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I was asked to ping @conda-forge/staged-recipes and so here I am doing that.

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I was asked to ping @conda-forge/staged-recipes and so here I am doing that.

ShixiangWang commented:
@conda-forge/staged-recipes Ready for review!

ShixiangWang commented:
@conda-forge-admin, please ping team

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-webservice.

I was asked to ping @conda-forge/staged-recipes and so here I am doing that.

ShixiangWang commented:
[@conda-forge/help-r](https://github.com/orgs/conda-forge/teams/help-r) ready for review.

ShixiangWang commented:
Dear @dpryan79, could you help review and approve this PR? :)

dbast commented on recipes/ucscxenashiny/meta.yaml:
```diff
@@ -0,0 +1,103 @@
+  # If this is a new build for the same version, increment the build number.
+  number: 0
+  # no skip
+
```
Can this be noarch as the others?

dbast commented:
the recipes look fine except:
* r recipe folders are normally prefixed with "r-".
* One of the recipes looks like it could be noarch: generic like the others
* the recipes contain lots of commented lines ... either remove those manually or use https://github.com/bgruening/conda_r_skeleton_helper to regenerate the recipes (this also automatically does the folder prefixing and noarch)

conda-forge-linter commented:
Hi! This is the friendly automated conda-forge-linting service.

I just wanted to let you know that I linted all conda-recipes in your PR (```recipes/r-ezcox```, ```recipes/r-forestmodel```, ```recipes/r-ucscxenashiny```, ```recipes/r-ucscxenatools```) and found it was in an excellent condition.

ShixiangWang commented:
@dbast Thanks for your comments. I have renamed the directories and removed comments. The linter suggested me to remove the `noarch` in `r-ucscxenashiny` as I added some required libraries only available in Linux. The two libraries are required in Linux to enable full features of this package.

```
  host:
    - gmp  # [linux]
    - mpfr  # [linux]
```
